src/runtime_src/core/tools/xbmgmt2/flash/flasher.cpp

```diff
@@ -229,15 +231,15 @@ Flasher::Flasher(unsigned int index) : mFRHeader{}
     //     }
     // }
     // else if (is_mfg)
-    //if (is_mfg)
-    //{
-    //    dev->read(MFG_REV_OFFSET, &mGoldenVer, sizeof(mGoldenVer));
-    //}
-    //else
-    //{
+    if (is_mfg)
+    {
+       dev->read(MFG_REV_OFFSET, &mGoldenVer, sizeof(mGoldenVer));
+    }
+    // else
+    // {
     //    std::cout << "ERROR: card not supported." << std::endl;
-    //}
-    m_device = dev; // Successfully initialized
+    // }
+    m_device = std::dynamic_pointer_cast<xrt_core::device>(dev); // Successfully initialized
```

Comment:
dynamic_cast.  Not sure I follow what's going on here?

Reply:
dev is of type pci_device whereas m_device is of type device on linux. Hance the dynamic cast

Reply:
Casting derived to base does not require dynamic_cast.  is it not directly assignable?